### PR TITLE
LG-3776: Adds LG-Chat!

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3776](https://jira.mongodb.org/browse/LG-3776)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: tools/lint/config/eslint.config.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ module.exports = {
       },
     },
     {
-      files: ['packages/**/*.spec.{ts,tsx}'],
+      files: ['chat/**/*.spec.{ts,tsx}', 'packages/**/*.spec.{ts,tsx}'],
```
Comment (TheSonOfThomp) [inline tools/lint/config/eslint.config.js:154]: I think this should also include tools (while we're touching this line)
https://jira.mongodb.org/browse/LG-4093

Comment (TheSonOfThomp) [reply tools/lint/config/eslint.config.js:154]: also: `'{chat,packages,tools}/**/*.spec.{ts,tsx}'`

File: tools/lint/src/eslint.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { LintFn } from './lint.types';
 
 const rootDir = process.cwd();
 const eslintConfigPath = path.resolve(__dirname, '../config/eslint.config.js');
-export const esLintExtensions = ['js', 'ts', 'tsx'];
+export const esLintExtensions = ['ts', 'tsx'];
```
Comment (TheSonOfThomp) [inline tools/lint/src/eslint.ts:10]: Are we sure we want to omit and lingering `.js` files?

File: tools/storybook-addon/src/main/index.ts [hunk 3]
```diff
@@ -14,8 +14,8 @@ export { previewHead } from './preview-head';
 
 // @ts-expect-error https://github.com/storybookjs/storybook/issues/23624
 export const stories: StorybookConfig['stories'] = findStories(
-  '../{packages,tools,stories}/**/*.stor@(y|ies).@(js|ts|md)?(x)',
-  '../{packages,tools}/*/node_modules',
+  '../{packages,tools,chat,stories}/**/*.stor@(y|ies).@(js|ts|md)?(x)',
+  '../{packages,tools,chat}/*/node_modules',
```
Comment (TheSonOfThomp) [inline tools/storybook-addon/src/main/index.ts:18]: Note for later: this should read from the `lg.json` file

File: tools/lint/config/eslint.config.js (hunk 1 above)
Comment (bruugey) [reply tools/lint/config/eslint.config.js]: 👍

File: tools/lint/src/eslint.ts (hunk 2 above)
Comment (bruugey) [reply tools/lint/src/eslint.ts:10]: i think so - only js files we should have are transpiled?

File: tools/storybook-addon/src/main/index.ts (hunk 3 above)
Comment (bruugey) [reply tools/storybook-addon/src/main/index.ts:18]: Make a ticket :)

File: .changeset/gorgeous-comics-notice.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+'@lg-tools/storybook-addon': minor
+'@lg-tools/storybook': minor
```
Comment (TheSonOfThomp) [inline .changeset/gorgeous-comics-notice.md:3]: The `/storybook` package is deprecated. We don't need to update it (I just haven't removed it yet)

Comment (bruugey) [reply .changeset/gorgeous-comics-notice.md]: Updated!

**Review by TheSonOfThomp — APPROVED** [review]

File: chat/avatar/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  "publishConfig": {
+    "registry": "https://artifactory.corp.mongodb.com/artifactory/api/npm/leafygreen-ui/"
+  },
+  "dependencies": {
+    "@leafygreen-ui/emotion": "^4.0.7",
+    "@leafygreen-ui/icon": "^11.22.2",
+    "@leafygreen-ui/lib": "^11.0.0",
+    "@leafygreen-ui/logo": "^8.0.8",
+    "@leafygreen-ui/palette": "^4.0.7",
+    "@leafygreen-ui/tokens": "^2.1.4"
```
Comment (TheSonOfThomp) [inline chat/avatar/package.json:29]: We should make sure all these are updated to the latest

File: chat/avatar/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# @lg-chat/avatar
+
+## 2.0.6
+
+### Patch Changes
```
Comment (TheSonOfThomp) [inline chat/avatar/CHANGELOG.md:5]: Should we make a note in the CHANGELOGs that previous to version X.Y.Z the package was private?

File: chat/avatar/package.json (hunk 5 above)
Comment (bruugey) [reply chat/avatar/package.json:29]: i cant run yarn run lg --latest at the root for this can i?

Comment (bruugey) [reply chat/avatar/package.json:29]: actually inclined to get this in and version nump in a follow up when we have chromatic

**Comment by TheSonOfThomp** [comment]
Noticed that [this line](https://github.com/mongodb/leafygreen-ui/blob/brooke/add-chat/.eslintignore#L7) should target all scope directories

**Comment by bruugey** [comment]
> Noticed that [this line](https://github.com/mongodb/leafygreen-ui/blob/brooke/add-chat/.eslintignore#L7) should target all scope directories. (Also, why are we `eslintignore`ing all of `lib`?)

Great question, dunno

File: chat/avatar/CHANGELOG.md (hunk 6 above)
Comment (bruugey) [reply chat/avatar/CHANGELOG.md:5]: They say "First public release" :)

**Comment by bruugey** [comment]
> Noticed that [this line](https://github.com/mongodb/leafygreen-ui/blob/brooke/add-chat/.eslintignore#L7) should target all scope directories. (Also, why are we `eslintignore`ing all of `lib`?)

We actually can rm that line bc i'm ignoring all js files

**Pull request merged**